api-server/ws/connection_manager.py

```diff
@@ -0,0 +1,51 @@
+from fastapi.websockets import WebSocket
+import anyio
+
+
+class Connection:
+    def __init__(self, id: str, group: str, websocket: WebSocket):
+        self.__id = id
+        self.__group = group
+        self.__websocket = websocket
+
+    async def send(self, data: dict):
+        await self.__websocket.send_json(data)
+
+    def get_group(self):
+        return self.__group
+
+    async def close_with_error(self, error_data: dict):
+        await self.send(error_data)
+        await self.__websocket.close()
+
+    def __eq__(self, other: "Connection"):
+        return self.__id == other.__id and self.__group == other.__group
+
+
+class ConnectionManager:
```

Comment:
Can you please explain `ConnectionManager`, `Connection` and `group` to me? Sorry if we discussed this before, I might have forgotten since it has been a while.

Reply:
A `Connection` is composed of the websocket connection and the player who is connecting. With it, one can send messages from the server to a select player.

`ConnectionManager` enables sending messages (broadcast) to all connections belonging to a specific `group`. In this case, `group` can refer to the session. For example, [here](https://github.com/rohanshiva/gitgame/pull/48/files#diff-86e706b25d7746fe55a6a7acbdc477711f2b95ff09538eb5b7e4eab35dda7e3dR110-R111) is where connections are added to the `ConnectionManager` and [here](https://github.com/rohanshiva/gitgame/pull/48/files#diff-86e706b25d7746fe55a6a7acbdc477711f2b95ff09538eb5b7e4eab35dda7e3dR126-R128) is where messages are broadcasted to all connections for a session.